deepmd/utils/path.py

```diff
@@ -44,6 +44,9 @@ def __new__(cls, path: str, mode: str = "r"):
             raise FileNotFoundError(f"{path} not found")
         return super().__new__(cls)
 
+    def __getnewargs__(self):
+        return (self.path, self.mode)
```

Comment:
I don't see any promise in this parent class that `self.path` and `self.mode` will be defined, so it's not self-consistent.

Reply:
Related: https://stackoverflow.com/a/54814371/9567349